plaso/parsers/text_plugins/apple_pstxt.py

```diff
@@ -0,0 +1,289 @@
+  _TIME = (
+      pyparsing.Word(pyparsing.nums, min=1, max=2).set_results_name('hours') +
+      pyparsing.Suppress(':') +
+      pyparsing.Word(pyparsing.nums, exact=2).set_results_name('minutes') +
+      pyparsing.one_of('AM PM').set_results_name('ampm')
+  )
+
+  _WEEKDAY = (
+      pyparsing.Word(pyparsing.alphas, exact=3).set_results_name('weekday') +
+      pyparsing.Word(pyparsing.nums, exact=2).set_results_name('hours') +
+      pyparsing.one_of('AM PM').set_results_name('ampm')
+  )
+
+  _HEADER = (
+      pyparsing.Literal('USER             UID PRSNA   PID  PPID        F  %CPU '
+                        '%MEM PRI NI      VSZ    RSS WCHAN    TT  STAT STARTED'
+                        '      TIME COMMAND') + pyparsing.LineEnd())
+
+  _COMMAND = pyparsing.OneOrMore(
+    pyparsing.Word(pyparsing.printables),
+    stop_on=pyparsing.LineEnd()).set_parse_action(' '.join)
+
+  _LOG_LINE = (
+      pyparsing.Word(pyparsing.alphanums + '_').set_results_name('user') +
+      pyparsing.Word(pyparsing.nums).set_results_name('user_identifier') +
+      pyparsing.Word(pyparsing.nums + '-').set_results_name('persona') +
+      pyparsing.Word(pyparsing.nums).set_results_name('process_identifier') +
+      pyparsing.Word(pyparsing.nums).set_results_name(
+        'parent_process_identifier') +
+      pyparsing.Word(pyparsing.alphanums).set_results_name('flags') +
+      pyparsing.Word(pyparsing.nums + '.').set_results_name('cpu') +
+      pyparsing.Word(pyparsing.nums + '.').set_results_name('memory') +
+      pyparsing.Word(pyparsing.nums).set_results_name('scheduling_priority') +
+      pyparsing.Word(pyparsing.nums).set_results_name('nice_value') +
+      pyparsing.Word(pyparsing.nums).set_results_name('virtual_size') +
+      pyparsing.Word(pyparsing.nums).set_results_name('resident_set_size') +
+      pyparsing.Word(pyparsing.nums + '-').set_results_name('wait_channel') +
+      pyparsing.Word(pyparsing.alphanums + '?').set_results_name(
+        'control_terminal_name') +
+      pyparsing.Word(pyparsing.alphanums).set_results_name(
+        'symbolic_process_state') +
+      pyparsing.Word(pyparsing.alphanums + ':').set_results_name('start_time') +
+      pyparsing.Word(pyparsing.nums + '.:').set_results_name('up_time') +
+      _COMMAND.set_results_name('command')
+  )
+
+  _LINE_STRUCTURES = [('log_line', _LOG_LINE), ('header', _HEADER)]
+
+  VERIFICATION_GRAMMAR = pyparsing.Suppress(_HEADER) + _LOG_LINE
+
+  def _ParseRecord(self, parser_mediator, key, structure):
+    """Parses a pyparsing structure.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfVFS.
+      key (str): name of the parsed structure.
+      structure (pyparsing.ParseResults): tokens from a parsed log line.
+
+    Raises:
+      ParseError: if the structure cannot be parsed.
+    """
+    # Retrieve the data from the file's metadata
+    self._SetEstimatedDate(parser_mediator)
```

Comment:
Move this into CheckRequiredFormat e.g. https://github.com/log2timeline/plaso/blob/main/plaso/parsers/text_plugins/syslog.py#L761